File: Patterns/Pattern 18.cpp
class Solution {
public:
    void pattern18(int n) {
        char chr = 'A' + n - 1;
        for(int i = 0; i < n; i++)
        {
            for(char ch = chr - i; ch <= chr; ch++)
            {
                cout << ch << " ";
            }
            cout << endl;
        }

    }
};

File: Patterns/Pattern 18.py
"""
📝 Question:
Print a reverse alphabet triangle.
Each row starts from a decreasing letter up to the last letter.

For n = 4 (last letter = D):
  D
  C D
  B C D
  A B C D
"""

# Pattern 18: Reverse Alphabet Triangle
# E
# D E
# C D E
# B C D E
# A B C D E

def pattern18(n):
    for i in range(n):
        start = chr(ord('A') + n - 1 - i)
        end = chr(ord('A') + n - 1)
        print(" ".join(chr(ord(start) + j) for j in range(i + 1)))

pattern18(5)
